consensus/src/types/primitives.rs

```diff
@@ -76,20 +77,34 @@ impl<const N: usize> ssz_rs::Deserialize for ByteVector<N> {
 
 impl<const N: usize> ssz_rs::SimpleSerialize for ByteVector<N> {}
 
+impl<const N: usize> serde::Serialize for ByteVector<N>
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let bytes = encode(self.inner.as_slice());
+        Ok(serializer.serialize_str(format!("0x{}", bytes).as_str())?)
+    }
+}
+
```

Comment:
1. Why do you need to implement a custom serialization for this struct?
2. Why did you choose to interfere with the format (i.e. why did you output as hex string, and added "0x"? a slice should be directly serialized)

Reply:
Ok I see now that you just updated their code, the idea was to remove unsafe functionality?

Reply:
It is very odd that they chose to implement this as they did.

Reply:
They created a custom deserialization for this struct that only deserialize from json format, which caused problems when needed to deserialize it on the Move code side (because we serialize this to bcs for size considerations).
So this custom serialize would serialize this as a string and then the deserialization works as we want.

Note that we could not modify the deserialize function's logic because it is used when fetching the data from the RPC.

Only modification to the deserialize function is to prevent panics when deserializing.